python/pyspark/ml/clustering.py

```diff
@@ -1156,6 +1159,216 @@ def getKeepLastCheckpoint(self):
+        Sets params for PowerIterationClustering.
+        """
+        kwargs = self._input_kwargs
+        return self._set(**kwargs)
+
+    @since("2.4.0")
+    def setK(self, value):
+        """
+        Sets the value of :py:attr:`k`.
+        """
+        return self._set(k=value)
+
+    @since("2.4.0")
+    def getK(self):
+        """
+        Gets the value of :py:attr:`k`.
+        """
+        return self.getOrDefault(self.k)
+
+    @since("2.4.0")
+    def setInitMode(self, value):
+        """
+        Sets the value of :py:attr:`initMode`.
+        """
+        return self._set(initMode=value)
+
+    @since("2.4.0")
+    def getInitMode(self):
+        """
+        Gets the value of `initMode`
+        """
+        return self.getOrDefault(self.initMode)
+
+    @since("2.4.0")
+    def setSrcCol(self, value):
+        """
+        Sets the value of :py:attr:`srcCol`.
+        """
+        return self._set(srcCol=value)
+
+    @since("2.4.0")
+    def getSrcCol(self):
+        """
+        Gets the value of :py:attr:`srcCol`.
+        """
+        return self.getOrDefault(self.srcCol)
+
+    @since("2.4.0")
+    def setDstCol(self, value):
+        """
+        Sets the value of :py:attr:`dstCol`.
+        """
+        return self._set(dstCol=value)
+
+    @since("2.4.0")
+    def getDstCol(self):
+        """
+        Gets the value of :py:attr:`dstCol`.
+        """
+        return self.getOrDefault(self.dstCol)
+
+    @since("2.4.0")
+    def assignClusters(self, dataset):
+        """
+        Run the PIC algorithm and returns a cluster assignment for each input vertex.
+
+        :param dataset:
+          A dataset with columns src, dst, weight representing the affinity matrix,
+          which is the matrix A in the PIC paper. Suppose the src column value is i,
+          the dst column value is j, the weight column value is similarity s,,ij,,
+          which must be nonnegative. This is a symmetric matrix and hence
+          s,,ij,, = s,,ji,,. For any (i, j) with nonzero similarity, there should be
+          either (i, j, s,,ij,,) or (j, i, s,,ji,,) in the input. Rows with i = j are
+          ignored, because we assume s,,ij,, = 0.0.
+
+        :return: A dataset that contains columns of vertex id and the corresponding cluster for
+                 the id. The schema of it will be:
+                   - id: Long
+                   - cluster: Int
+        """
+        weightCol = None
+        w = None
+        if (self.isDefined(self.weightCol)):
+            weightCol = self.getWeightCol()
+        if (weightCol is None or len(weightCol) == 0):
+            w = lit(1.0)
+        else:
+            w = col(weightCol).cast(DoubleType())
+        srcCol = self.getSrcCol()
+        dstCol = self.getDstCol()
+        df = dataset.select(col(srcCol).cast(LongType()), col(dstCol).cast(LongType()), w)
+        data = df.rdd.map(lambda x: (x[0], x[1], x[2]))
+        algorithm = MLlibPowerIterationClustering()
```

Comment:
I think here should wrap `ml.clustering.PowerIterationClustering`, not the one in `pyspark.mllib`. So that it can make sure behavior consistency with scala api.

Reply:
@WeichenXu123 Thank you very much for your comment. Will change now.